RTClib.h

```diff
@@ -57,204 +57,17 @@ class TimeSpan;
-  */
-  bool operator>=(const DateTime &right) const { return !(*this < right); }
-  bool operator==(const DateTime &right) const;
-
-  /*!
-      @brief  Test if two DateTime objects are not equal.
-      @warning if one or both DateTime objects are invalid, returned value is
-        meaningless
-      @see use `isValid()` method to check if DateTime object is valid
-      @param right DateTime object to compare
-      @return True if the two objects are not equal, false if they are
-  */
-  bool operator!=(const DateTime &right) const { return !(*this == right); }
-
-protected:
-  uint8_t yOff; ///< Year offset from 2000
-  uint8_t m;    ///< Month 1-12
-  uint8_t d;    ///< Day 1-31
-  uint8_t hh;   ///< Hours 0-23
-  uint8_t mm;   ///< Minutes 0-59
-  uint8_t ss;   ///< Seconds 0-59
-};
-
-/**************************************************************************/
-/*!
-    @brief  Timespan which can represent changes in time with seconds accuracy.
-*/
-/**************************************************************************/
-class TimeSpan {
-public:
-  TimeSpan(int32_t seconds = 0);
-  TimeSpan(int16_t days, int8_t hours, int8_t minutes, int8_t seconds);
-  TimeSpan(const TimeSpan &copy);
-
-  /*!
-      @brief  Number of days in the TimeSpan
-              e.g. 4
-      @return int16_t days
-  */
-  int16_t days() const { return _seconds / 86400L; }
-  /*!
-      @brief  Number of hours in the TimeSpan
-              This is not the total hours, it includes the days
-              e.g. 4 days, 3 hours - NOT 99 hours
-      @return int8_t hours
-  */
-  int8_t hours() const { return _seconds / 3600 % 24; }
-  /*!
-      @brief  Number of minutes in the TimeSpan
-              This is not the total minutes, it includes days/hours
-              e.g. 4 days, 3 hours, 27 minutes
-      @return int8_t minutes
-  */
-  int8_t minutes() const { return _seconds / 60 % 60; }
-  /*!
-      @brief  Number of seconds in the TimeSpan
-              This is not the total seconds, it includes the days/hours/minutes
-              e.g. 4 days, 3 hours, 27 minutes, 7 seconds
-      @return int8_t seconds
-  */
-  int8_t seconds() const { return _seconds % 60; }
-  /*!
-      @brief  Total number of seconds in the TimeSpan, e.g. 358027
-      @return int32_t seconds
-  */
-  int32_t totalseconds() const { return _seconds; }
-
-  TimeSpan operator+(const TimeSpan &right);
-  TimeSpan operator-(const TimeSpan &right);
-
-protected:
-  int32_t _seconds; ///< Actual TimeSpan value is stored as seconds
-};
+#if (ARDUINO >= 100)
+#include <Arduino.h> // capital A so it is error prone on case-sensitive filesystems
+// Macro to deal with the difference in I2C write functions from old and new
+// Arduino versions.
+#define _I2C_WRITE write ///< Modern I2C write
+#define _I2C_READ read   ///< Modern I2C read
+#else
+#include <WProgram.h>
+#define _I2C_WRITE send   ///< Legacy I2C write
+#define _I2C_READ receive ///< legacy I2C read
+#endif
```

Comment:
This was originally in RTClib.cpp. Keeping it there would limit the pollution to the global namespace.

Reply:
Fixed